### PR TITLE
Improve peason r function

### PR DESCRIPTION
**Pull request by murilodag**
Add documentation header
Comment out console.log
Reduce function size
Calculate numerator only when denominatior isn't zero

### CODE REVIEW

File: src/inference/inference.js [hunk 1]
```diff
@@ -124,40 +124,72 @@ function independentTTest (numbers, marginOfError = 0.05, twoTailed = true) {
...
+
     data.push({
       right: rightData[i],
       left: leftData[i]
     });
   }
+  return data;
+}
+
+function arraySum (array) {
```
Comment (johnpaulada) [inline src/inference/inference.js]: You can just import the `sum` function on the `src/basic/basic.js` module. 😄

File: src/inference/inference.js [hunk 2]
```diff
@@ -124,40 +124,72 @@ function independentTTest (numbers, marginOfError = 0.05, twoTailed = true) {
...
+  return array.reduce((sum, current) => current + sum, 0)
+}
+
+function arraySquaredSum(array) {
```
Comment (johnpaulada) [inline src/inference/inference.js]: Maybe you can move this to the basic module as well?

**Review by johnpaulada — CHANGES_REQUESTED** [review]
I left some comments for the code. Kindly check them out. They're just small changes. Thanks!

File: src/inference/inference.js (hunk 2 above)
Comment (murilodag) [reply src/inference/inference.js]: Moved.

File: src/inference/inference.js (hunk 1 above)
Comment (murilodag) [reply src/inference/inference.js]: I missed that, thanks

**Comment by johnpaulada** [comment]
Oh sorry I didn't notice the changes. Will merge it now. Great job! 🎉

**Pull request merged**